cmd/nerdctl/container_run_network_linux_test.go

```diff
@@ -493,6 +493,21 @@ func TestRunContainerWithMACAddress(t *testing.T) {
 	}
 }
 
+func TestHostsFilePermissions(t *testing.T) {
+	base := testutil.NewBase(t)
+	base.Cmd("run", "--rm", testutil.CommonImage,
+		"sh", "-c", "echo > /etc/hosts").AssertOK()
+	base.Cmd("run", "--rm", "-v", "/etc/hosts:/etc/hosts", "--network", "host", testutil.CommonImage,
+		"sh", "-c", "echo > /etc/hosts").AssertOK()
+	base.Cmd("run", "--rm", "-v", "/etc/hosts:/etc/hosts:ro", "--network", "host", testutil.CommonImage,
+		"sh", "-c", "echo > /etc/hosts").AssertFail()
+
+	// The mount of /etc/host is incompatible with docker
```

Comment:
Could you explain this?

Reply:
> Could you explain this?

HI @AkihiroSuda 

In docker, the /etc/hosts is mounted by the writable: https://github.com/moby/moby/blob/v24.0.7/container/container_unix.go#L113,

But in nerdctl, it's defined as read-only in the containerd's code:

https://github.com/containerd/nerdctl/blob/v1.7.1/pkg/containerutil/container_network_manager.go#L395
https://github.com/containerd/containerd/blob/v1.7.11/oci/spec_opts.go#L514
Because it cannot be changed, so keep it read-only.

Reply:
> Should I change it to writable the same as docker ?

Yes, if possible

Reply:
Thanks :-)
I'll reopen a PR to fix it.